runtimes/go/appruntime/shared/reqtrack/trace_stream.go

```diff
@@ -46,9 +47,17 @@ func (l *lazyTraceInit) Logger() trace2.Logger {
 	return l.log
 }
 
+func (l *lazyTraceInit) WaitForStreamSent() {
+	l.sent.Wait()
```

Comment:
This will return immediately if `l.initStream` hasn't been called. Is that intended?

Reply:
initStream is always called before WaitForStream; the only call site is:

```go
		op.trace.MarkDone()

		if blockOnTraceSend {
			op.trace.WaitForStreamSent()
		}
```

Where `MarkDone` is:

```go
func (l *lazyTraceInit) MarkDone() {
	l.log.MarkDone()
	l.initStream()
}
```